daemon/daemon_unix.go

```diff
@@ -576,6 +576,22 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 		}
 	}
 
+	if hostConfig.StorageOpt != nil && daemon.GraphDriverName(runtime.GOOS) == "overlay2" {
+		_, exist := hostConfig.StorageOpt["size"]
+		if exist {
+			if hostConfig.Privileged {
+				warnings = append(warnings, "filesystem quota for overlay2 over ext4 or xfs can't take affect with privileged container")
```

Comment:
Actually, that is not true for xfs. xfs does not use the generic vfs quota enforcement and enforces quotas also for privileged user (and for user with CAP_SYS_RESOURCE)